include/onnxruntime/core/providers/providers.h

```diff
@@ -5,9 +5,11 @@
 
 namespace onnxruntime {
 class IExecutionProvider;
+struct SessionOptions;
 
 struct IExecutionProviderFactory {
   virtual ~IExecutionProviderFactory() = default;
-  virtual std::unique_ptr<IExecutionProvider> CreateProvider() = 0;
+  virtual std::unique_ptr<IExecutionProvider> CreateProvider(
+      const SessionOptions* options = nullptr) = 0;
 };
```

Comment:
What's the reason for this change?

Reply:
Currently, How Ep created is decoupled with `SessionOptions`, so we don't know anything about what's the related configuration value when an EP was created.
However, for some EPs, they need to know if some configs were conflict with its unique features. Like CUDA/DML, that's why they have to hardcode into `inference_session.cc`.

For XNNPACK, we want to check if ort's intra-threadpool spinning and thread-pool size were compatible with XNNPACK EP, would it be best fit to check it when creating a EP?

Reply:
Given where we are in the release cycle, my suggestion would be to keep this PR minimal for now and not change the EP interface. We can document the suggested setup for xnnpack usage and also include that in the release notes so it has high visibility.

In the next release we can figure out what we want to do with SessionOptions and threadpool unification. 

@pranavsharma do you agree?

Reply:
Makes sense. Let's defer the EP interface changes for now.